File: src/main/java/com/matamercer/microblog/models/repositories/UserRepository.java
package com.matamercer.microblog.models.repositories;

import com.matamercer.microblog.models.entities.AuthenticationProvider;
import com.matamercer.microblog.models.entities.User;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {
    Optional<User> findByUsername(String username);
    Optional<User> findByEmail(String email);
    Optional<User> findByoAuth2IdAndAuthenticationProvider(String oAuth2Id, AuthenticationProvider authenticationProvider);
}

File: src/main/java/com/matamercer/microblog/models/repositories/UserRepository.kt
package com.matamercer.microblog.models.repositories

import com.matamercer.microblog.models.entities.AuthenticationProvider
import com.matamercer.microblog.models.entities.User
import org.springframework.data.repository.CrudRepository

interface UserRepository : CrudRepository<User, Long> {
    fun findByUsername(username: String?): User?
    fun findByEmail(email: String?): User?
    fun findByoAuth2IdAndAuthenticationProvider(
        oAuth2Id: String?,
        authenticationProvider: AuthenticationProvider?
    ): User?
}